File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        count = {}
        for n in nums:
            if n in count:
                return n
            count[n] = 1
        
        current = 0
        next = nums[current]

        slow = nums[0]
        fast = nums[0]
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            if slow == fast:
                return nums[slow]

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-0.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int res = 0;
        for (int l = 0; l < nums.length; l++) {
            int count = 0, r = l;
            while (r < nums.length) {
                if (nums[r] == 0) {
                    if (count == k) break;
                    count++;
                }
                r++;
            }
            res = Math.max(res, r - l);
        }
        return res;
    }
}